### ISSUE TITLE
Deprecate using 'tagname' with the LinkTo component

### ISSUE THREAD

**Issue by MelSumner** [emberjs/rfcs#767]
In general, the LinkTo component is useful, but there seems to be a leftover from the past. 

![image](https://user-images.githubusercontent.com/4587451/134492544-214d6cb3-fe8a-4933-b688-61dec2a97fa3.png)

This should be deprecated because it's confusing and also it lends itself to accessibility errors. The fewer footguns we provide in the accessibility space, the better. 

If this is already deprecated, then the API docs need to be updated.

**Comment by bertdeblock** [comment]
This should already be the case I think:
https://github.com/emberjs/ember.js/blob/57907a4d7d91aa58b3b3ac1258e639ce221f9492/packages/@ember/-internals/glimmer/lib/components/link-to.ts#L432-L445

**Comment by chriskrycho** [comment]
@MelSumner do you mind replacing the image here with the actual source from the docs? (Ironically, the image is inaccessible! 😅)

Also, for anyone curious, the current docs are [here](https://api.emberjs.com/ember/3.28/classes/Ember.Templates.components/methods/LinkTo?anchor=LinkTo), and as @bertdeblock noted, this was deprecated as part of #707 – the deprecation info is [here](https://deprecations.emberjs.com/v3.x#toc_ember-built-in-components-legacy-arguments).

**Comment by locks** [comment]
Closing as resolved! I was looking at LinkTo RFCs because I forgot about the Built-in components one 🤦
